cmd/dagger/mcp.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"dagger.io/dagger/querybuilder"
+	"github.com/dagger/dagger/dagql/idtui"
+	"github.com/dagger/dagger/engine/client"
+	"github.com/spf13/cobra"
+)
+
+var (
+	mcpStdio   bool
+	mcpSseAddr string
+)
+
+func init() {
+	mcpCmd.PersistentFlags().BoolVar(&mcpStdio, "stdio", true, "Use standard input/output for communicating with the MCP server")
+	mcpCmd.PersistentFlags().StringVar(&mcpSseAddr, "sse-addr", "", "Address of the MCP SSE server (no SSE server if empty)")
+}
+
+var mcpCmd = &cobra.Command{
+	Use:   "mcp [options]",
+	Short: "Expose a dagger module as an MCP server",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SetContext(idtui.WithPrintTraceLink(cmd.Context(), true))
+		return withEngine(cmd.Context(), client.Params{}, mcpStart)
+	},
+	Hidden: true,
+	Annotations: map[string]string{
+		"experimental": "true",
+	},
+}
+
+// dagger -m github.com/org/repo mcp key1=val1 key2=val2
```

Comment:
what do you see these keys and values being passed to? are they module-level flags?

Reply:
My bad, this is more of a TODO of how to handle module constructors. Forgot to mention it in the description.

Reply:
if it's possible they should retain the syntax of `call`, like --module-constructor=value